homeassistant/components/sensor/alpha_vantage.py

```diff
@@ -0,0 +1,125 @@
+"""
+Stock market information from Alpha Vantage.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.alpha_vantage/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_API_KEY
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['alpha_vantage==1.3.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_CLOSE = 'close'
+ATTR_HIGH = 'high'
+ATTR_OPEN = 'open'
+ATTR_LOW = 'low'
+ATTR_VOLUME = 'volume'
+
+CONF_ATTRIBUTION = "Stock market information provided by Alpha Vantage."
+CONF_SYMBOLS = 'symbols'
+
+DEFAULT_SYMBOL = 'GOOGL'
+
+ICON = 'mdi:currency-usd'
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Optional(CONF_SYMBOLS, default=[DEFAULT_SYMBOL]):
+        vol.All(cv.ensure_list, [cv.string]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Alpha Vantage sensor."""
+    from alpha_vantage.timeseries import TimeSeries
+
+    api_key = config.get(CONF_API_KEY)
+    symbols = config.get(CONF_SYMBOLS)
+
+    timeserie = TimeSeries(key=api_key)
+
+    dev = []
+    for symbol in symbols:
+        try:
+            data = AlphaVantageData(timeserie, symbol)
```

Comment:
If each data instance will only report one symbol, and the same goes for the sensor instance, I don't think there's a benefit of separating the data fetching from the entity class.